olive/evaluator/olive_evaluator.py

```diff
@@ -334,8 +334,11 @@ def _inference(
                 # convert to dict of torch tensor
                 result = {name: torch.Tensor(res[i]) for i, name in enumerate(output_names)}
             outputs = post_func(result) if post_func else result
-            preds.extend(outputs.tolist())
-            targets.extend(labels.data.tolist())
+            # keep as numpy or torch arrays
+            preds.append(outputs.cpu())
```

Comment:
the append will make the list as a single item. That's the why the extend is used originally.

Reply:
torch.cat at line 340 and 341 does the same thing as noted in the PR description